oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStoreBuilder.java

```diff
@@ -534,13 +533,13 @@ public DiffCache getDiffCache(int clusterId) {
      * @return this
      */
     public T setBlobStore(BlobStore blobStore) {
-        this.blobStoreSupplier = ofInstance(blobStore);
+        this.blobStoreSupplier = () -> blobStore;
         return thisBuilder();
     }
 
     public BlobStore getBlobStore() {
         if (blobStoreSupplier == null) {
-            blobStoreSupplier = ofInstance(new MemoryBlobStore());
+            blobStoreSupplier = () -> new MemoryBlobStore();
```

Comment:
same issue could happen here, we need to create an instance of `new MemoryBlobStore` at instance level and pass that value in the `Supplier` lambda expression.